open-budget-il/main.py

```diff
@@ -708,6 +712,58 @@ def get_query(self):
 
         return results
 
+class AutocompleteApi(GenericApi):
+
+    def do_paging(self):
+        return False
+
+    def get_querystr(self):
+        try:
+            return self.request.get('q')
+        except:
+            qs = self.request.query_string
+            if 'q=' in qs:
+                qs = qs[qs.index('q=')+2:]
+                qs = qs.split('&')[0]
+                return urllib.unquote(qs).decode('windows-1255')
+            else:
+                return None
+
+    def key(self):
+        queryString = self.get_querystr()
+        return "AutocompleteApi:%s" % (queryString)
+
+    def get_query(self):
+        queryString = self.get_querystr()
+        autocompleteIndex = search.Index(name="autocomplete")
```

Comment:
True, 
I went with your implementation of adding the tokens inside the document which is much more efficient